presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -200,6 +210,12 @@ else if (expression instanceof IsNullPredicate) {
             else if (expression instanceof IsNotNullPredicate) {
                 validateShowStatsWhereExpression(((IsNotNullPredicate) expression).getValue(), node);
             }
+            else if (expression instanceof InPredicate) {
+                validateShowStatsWhereExpression(((InPredicate) expression).getValue(), node);
+            }
+            else if (expression instanceof BetweenPredicate) {
+                validateShowStatsWhereExpression(((BetweenPredicate) expression).getValue(), node);
+            }
```

Comment:
maybe you could use `com.facebook.presto.sql.util.AstUtils#preOrder` and check if all nodes are `com.facebook.presto.util.MorePredicates#isInstanceOfAny(ALLOWED_SHOW_STATS_WHERE_EXPRESSION_TYPES)`?